src/MeshInertiaCalculator.cc

```diff
@@ -0,0 +1,223 @@
+      integral[5] += crosses[i].Y() * f3[i].Y();
+      integral[6] += crosses[i].Z() * f3[i].Z();
+      integral[7] += crosses[i].X() *
+        (y0 * g0[i].X() + y1 * g1[i].X() + y2 * g2[i].X());
+      integral[8] += crosses[i].Y() *
+        (z0 * g0[i].Y() + z1 * g1[i].Y() + z2 * g2[i].Y());
+      integral[9] += crosses[i].Z() *
+        (x0 * g0[i].Z() + x1 * g1[i].Z() + x2 * g2[i].Z());
+  }
+
+  for (int i = 0; i < 10; ++i)
+  {
+      integral[i] *= coefficients[i];
+  }
+
+  // Accumulate the result and add it to MassMatrix object of gz::math
+  double volume = integral[0];
+  double mass = volume * _density;
+  _centreOfMass.SetX(integral[1] / mass);
+  _centreOfMass.SetY(integral[2] / mass);
+  _centreOfMass.SetZ(integral[3] / mass);
+  gz::math::Vector3d ixxyyzz = gz::math::Vector3d();
+  gz::math::Vector3d ixyxzyz = gz::math::Vector3d();
+
+  // Diagonal Elements of the Mass Matrix
+  ixxyyzz.X() = (integral[5] + integral[6] - mass *
+                (_centreOfMass.Y() * _centreOfMass.Y() +
+                _centreOfMass.Z() * _centreOfMass.Z()));
+  ixxyyzz.Y() = (integral[4] + integral[6] - mass *
+                (_centreOfMass.Z() * _centreOfMass.Z() +
+                _centreOfMass.X() * _centreOfMass.X()));
+  ixxyyzz.Z() = integral[4] + integral[5] - mass *
+                (_centreOfMass.X() * _centreOfMass.X() +
+                _centreOfMass.Y() * _centreOfMass.Y());
+
+  // Off Diagonal Elements of the Mass Matrix
+  ixyxzyz.X() = -(integral[7] - mass * _centreOfMass.X() * _centreOfMass.Y());
+  ixyxzyz.Y() = -(integral[8] - mass * _centreOfMass.Y() * _centreOfMass.Z());
+  ixyxzyz.Z() = -(integral[9] - mass * _centreOfMass.X() * _centreOfMass.Z());
+
+  // Set the values in the MassMatrix object
+  _massMatrix.SetMass(mass);
+  _massMatrix.SetDiagonalMoments(ixxyyzz * _density);
+  _massMatrix.SetOffDiagonalMoments(ixyxzyz * _density);
+}
+
+//////////////////////////////////////////////////
+std::optional<gz::math::Inertiald> MeshInertiaCalculator::operator()
+  (sdf::Errors& _errors,
+  const sdf::CustomInertiaCalcProperties& _calculatorParams)
+{
+  const gz::common::Mesh *mesh = nullptr;
+  const double density = _calculatorParams.Density();
+  const sdf::Mesh sdfMesh = _calculatorParams.Mesh();
+  auto fullPath = asFullPath(sdfMesh.Uri(), sdfMesh.FilePath());
+
+  if (fullPath.empty())
+  {
+    gzerr << "Mesh geometry missing uri" << std::endl;
+    _errors.push_back({sdf::ErrorCode::URI_INVALID,
+        "Attempting to load the mesh but the URI seems to be incorrect"});
+    return std::nullopt;
+  }
+
+  // Load the Mesh
+  gz::common::MeshManager *meshManager = gz::common::MeshManager::Instance();
+  mesh = meshManager->Load(fullPath);
+  std::vector<Triangle> meshTriangles;
+  gz::math::MassMatrix3d meshMassMatrix;
+  gz::math::Pose3d centreOfMass;
+
+  // Create a list of Triangle objects from the mesh vertices and indices
+  GetMeshTriangles(meshTriangles, mesh);
```

Comment:
add `this->` when calling member functions, i.e. 
`this->GetMeshTriangles(meshTriangles, mesh);` and 
`this->CalculateMassProperties(meshTriangles, density, meshMassMatrix, centreOfMass);`

Reply:
Used `this->` for calling member functions in [b46c2b5](https://github.com/gazebosim/gz-sim/pull/2061/commits/b46c2b559b939d0112ccfa1b71bed22807c43703)